src/vs/workbench/browser/positronDataGrid/components/dataGridWaffle.tsx

```diff
@@ -71,50 +72,78 @@ export const DataGridWaffle = forwardRef<HTMLDivElement>((_: unknown, ref) => {
 		return () => disposableStore.dispose();
 	}, [services.configurationService, context.instance]);
 
-	// Layout useEffect.
-	useEffect(() => {
-		// Set the initial width and height.
-		setWidth(dataGridWaffleRef.current.offsetWidth);
-		setHeight(dataGridWaffleRef.current.offsetHeight);
+	// Automatic layout useLayoutEffect.
+	useLayoutEffect(() => {
+		// Wait for the data grid waffle to be mounted.
+		if (!dataGridWaffleRef.current) {
+			return;
+		}
 
 		/**
-		 * Sets the screen size.
+		 * Sets the size.
 		 * @returns A Promise<void> that resolves when the operation is complete.
 		 */
-		const setScreenSize = async (width: number, height: number) => {
-			// Set the screen size.
+		const setSize = async (width: number, height: number) => {
+			// Set the width and height in this component.
+			setWidth(width);
+			setHeight(height);
+
+			// Set the size in the data grid instance.
 			await context.instance.setSize(width, height);
 		};
 
-		// Set the initial screen size.
-		setScreenSize(
-			dataGridWaffleRef.current.offsetWidth,
-			dataGridWaffleRef.current.offsetHeight
+		// ResizeObserver does not work well on elements inside secondary Electron windows. This causes issues
+		// with the data grid's automatic layout feature, which relies on knowing when the size of the data grid
+		// waffle changes. See https://github.com/posit-dev/positron/issues/8695. This workaround ensures that the
+		// initial data grid waffle's size is updated when the component is mounted.
+		DOM.getWindow(dataGridWaffleRef.current).requestAnimationFrame(async () =>
+			await setSize(dataGridWaffleRef.current.offsetWidth, dataGridWaffleRef.current.offsetHeight)
```

Comment:
Suggestion to change the comment to be a bit more explicit about why we are using `requestAnimationFrame`. This change is more for me if I ever have to look at this again 😂 

```suggestion
		// ResizeObserver does not work well on elements inside secondary Electron windows. This causes issues
		// with the data grid's automatic layout feature, which relies on knowing when the size of the data grid
		// waffle changes. See https://github.com/posit-dev/positron/issues/8695. Using `requestAnimationFrame`
		// ensures that the initial size of the data grid waffle is set when the component is mounted.
		DOM.getWindow(dataGridWaffleRef.current).requestAnimationFrame(async () =>
			await setSize(dataGridWaffleRef.current.offsetWidth, dataGridWaffleRef.current.offsetHeight)
```